target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -142,6 +145,9 @@ ipq40xx_setup_macs()
 		lan_mac=$(cat /sys/firmware/mikrotik/hard_config/mac_base)
 		label_mac="$lan_mac"
 		;;
+	openmesh,a42)
+		label_mac="$(mtd_get_mac_binary "0:ART" 0x0)"
```

Comment:
If its in binary, shouldn't NVMEM work?

That would avoid the need for preinit setting as well

Reply:
Ok, I can have a look. But only a additional patch on top of this one. The integration of the partitions actually happens via bootargs (mtdparts) and not via devicetree. So we need now some placeholder entries in the DTS - similar to what is done for the openmesh/plasmacloud devices in ath79.

Reply:
You can actually avoid defining fixed-partitions and just add a node for ART as there is now NVMEM support for dynamic partition parsers:
https://github.com/openwrt/openwrt/commit/1a9ee367343edce263f82cc91a49d796c9d45ea3

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/Documentation/devicetree/bindings/mtd/partitions?h=v6.1-rc2&id=118f3fbe517f49e17877f34fd677f7374970d92e

Reply:
Thanks for the info. I have now switched the DTS to the shallow `partition-*` approach